pycqed/measurement/waveform_control/pulse_library.py

```diff
@@ -113,6 +113,188 @@ def hashables(self, tstart, channel):
         return hashlist
 
 
+class GaussianFilteredPiecewiseConstPulse(pulse.Pulse):
+    """
+    The base class for different Gaussian-filtered piecewise constant pulses.
+
+    To avoid clipping of the Gaussian-filtered rising and falling edges, the
+    pulse should start and end with zero-amplitude buffer segments.
+
+    Args:
+        name (str): The name of the pulse, used for referencing to other pulses
+            in a sequence. Typically generated automatically by the `Segment`
+            class.
+        element_name (str): Name of the element the pulse should be played in.
+        channels (list of str): Channel names this pulse is played on
+        lengths (list of list of float): For each channel, a list of the
+            lengths of the pulse segments. Must satisfy
+            `len(lengths) == len(channels)`.
+        amplitudes (list of list of float): The amplitudes of all pulse
+            segments. The shape must match that of `lengths`.
+        gaussian_filter_sigma (float): The width of the gaussian filter sigma
+            of the pulse.
+        codeword (int or 'no_codeword'): The codeword that the pulse belongs in.
+            Defaults to 'no_codeword'.
+    """
+
+    def __init__(self, name, element_name, channels, lengths, amplitudes,
+                 gaussian_filter_sigma=0, codeword='no_codeword', **kw):
+        self.name = name
+        self.element_name = element_name
+        self.codeword = codeword
+        self.channels = channels
+        self._t0 = None
+
+        self.lengths = lengths
+        self.amplitudes = amplitudes
+        self.gaussian_filter_sigma = gaussian_filter_sigma
```

Comment:
Even if this is a base class that is not directly used as a pulse, shouldn't the parameters still be specified in the classmethod pulse_params? Then, they would also be handled automatically by the init of the base pulse class, instead of handling them manually here in this init.

Reply:
This I will do, yes.